apps/dav/lib/Migration/BuildSocialSearchIndexBackgroundJob.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * @copyright 2020 Matthias Heinisch <nextcloud@matthiasheinisch.de>
+ *
+ * @author Matthias Heinisch <nextcloud@matthiasheinisch.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\DAV\Migration;
+
+use OC\BackgroundJob\QueuedJob;
+use OCA\DAV\CardDAV\CardDavBackend;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\BackgroundJob\IJobList;
+use OCP\IDBConnection;
+use OCP\ILogger;
+
+class BuildSocialSearchIndexBackgroundJob extends QueuedJob {
+
+	/** @var IDBConnection */
+	private $db;
+
+	/** @var CardDavBackend */
+	private $davBackend;
+
+	/** @var ILogger */
+	private $logger;
+
+	/** @var IJobList */
+	private $jobList;
+
+	/** @var ITimeFactory */
+	private $timeFactory;
+
+	/**
+	 * @param IDBConnection $db
+	 * @param CardDavBackend $davBackend
+	 * @param ILogger $logger
+	 * @param IJobList $jobList
+	 * @param ITimeFactory $timeFactory
+	 */
+	public function __construct(IDBConnection $db,
+								CardDavBackend $davBackend,
+								ILogger $logger,
+								IJobList $jobList,
+								ITimeFactory $timeFactory) {
+		$this->db = $db;
+		$this->davBackend = $davBackend;
+		$this->logger = $logger;
+		$this->jobList = $jobList;
+		$this->timeFactory = $timeFactory;
+	}
+
+	public function run($arguments) {
+		$offset = $arguments['offset'];
+		$stopAt = $arguments['stopAt'];
+
+		$this->logger->info('Indexing social profile data (' . $offset .'/' . $stopAt . ')');
+
+		$offset = $this->buildIndex($offset, $stopAt);
+
+		if ($offset >= $stopAt) {
+			$this->logger->info('All contacts with social profiles indexed');
+		} else {
+			$this->jobList->add(self::class, [
+				'offset' => $offset,
+				'stopAt' => $stopAt
+			]);
+			$this->logger->info('New social profile indexing job scheduled with offset ' . $offset);
+		}
+	}
+
+	/**
+	 * @param int $offset
+	 * @param int $stopAt
+	 * @return int
+	 */
+	private function buildIndex($offset, $stopAt) {
+		$startTime = $this->timeFactory->getTime();
+
+		// get contacts with social profiles
+		$query = $this->db->getQueryBuilder();
+		// TODO: return contacts with multiple social profiles only once
+		// FIXME: distinct seems only to return the first parameter?
+		// $query->selectDistinct('c.addressbookid', 'c.uri', 'c.carddata')
```

Comment:
any idea how the distinct query can be applied here?

Reply:
Why do you want to distinct? We need to update all entries?

```suggestion
		$query->select('c.addressbookid', 'c.uri', 'c.carddata')
			->selectDistinct('c.id')
```

but since its the id column you dont select duplicates anyway, so should be fine without it.

Reply:
Well, I am looking in the `cards_properties` for socialprofiles and then take the respective contacts from `cards` (so that I have only the ones with social profiles). But thinking about that now again, I guess the `card-properties` are in fact the indexed parameters, so that I cannot search in there and must take all contacts?

Reply:
I have changed that now, see 69f3dca6f5